boa/contracts/abi/__init__.py

```diff
@@ -1,28 +1,66 @@
-from typing import Any, Union
+from collections import defaultdict
+from typing import Any, Callable, Iterable, TypeVar, Union
 
 from boa.environment import Address
 
+T = TypeVar("T")
```

Comment:
i think typevars are overkill here but in any case -- move them closer to where they are used!